### DISCUSSION TITLE
Why input of rgb_to_lab and output of lab_to_rgb are not the same?

### DISCUSSION THREAD

**Discussion by hminle** [kornia/kornia#1144] [Questions] [answered]
Hi Kornia team,

I test `rbg_to_lab` and `lab_to_rgb` with these lines of code:
![image](https://user-images.githubusercontent.com/16201681/124204068-31d1df80-daac-11eb-8a27-a45df132d842.png)

The `input` and `output_rgb` are supposed to be the same, but it is not the case.
I checked the document but did not see any special preprocessing method for input/output of those conversion functions.

So why they are not the same?

Thanks.

**Comment by shijianjian** [comment]
After a quick glancing, I suspect it is related to those ```clamp``` functions.

@cceyda, can you take a look?

**Comment by cceyda** [comment]
Yes @shijianjian it is because of clipping done during `lab_to_rgb`. This is according to conversion standarts [[ref](https://www.w3.org/Graphics/Color/srgb)]
![image](https://user-images.githubusercontent.com/15624271/124231313-110c9700-db4b-11eb-9f83-ec09bde6d28a.png)

@hminle you can check our test for this exact thing [here](https://github.com/kornia/kornia/blob/25cb90b1775fa2244790cd8f46c6120f68106c6e/test/color/test_lab.py#L88) precision is within 1e-4. Let me know if this answers your question 😄

**Comment by edgarriba** [comment]
@cceyda @shijianjian should we have `clip==False` in `lab_to_rgb` ?

**Comment by cceyda** [comment]
@edgarriba This might lead to some problems if someone tries to save an image or pass it onto another (kornia) function that expects things in 0,1 range. Imo `clip=True` is a safer default.

**Comment by hminle** [comment, marked as answer]
Hi @cceyda ,

I found another test that solve the issue:
https://github.com/kornia/kornia/blob/25cb90b1775fa2244790cd8f46c6120f68106c6e/test/color/test_lab.py#L230

Thank you, I got it.

**Reply by hminle** [reply]
By the way, it would be better if we can say clearly about the range of input and output of those functions.

Take the function `kornia.color.rgb_to_lab` as an example, its input is RGB in the range 0-1, but its output is Lab space, which has L channel in the range 0-100, and a & b channel in the range of -100 to 100.

**Reply by edgarriba** [reply]
@hminle sure - we are now polishing our docs to improve usability

**Reply by cceyda** [reply]
@hminle actually the ranges are this:
L => (0, 100)
a, b => (-128, 128)

We are also considering maybe returning lab values scaled to [0,1], maybe through an scale=True option? or depending on dtype? Share if you have opinions on if that makes sense

**Reply by hminle** [reply]
Thank @cceyda for correcting me.
I see some colour lib can have `a, b` in the range of -100 to 100 too.
![image](https://user-images.githubusercontent.com/16201681/124300252-5a012300-db2c-11eb-8b9d-79f41c098442.png)
https://en.wikipedia.org/wiki/CIELAB_color_space
![image](https://user-images.githubusercontent.com/16201681/124300511-a8aebd00-db2c-11eb-8a5d-5f702dbeb7e7.png)

https://colour.readthedocs.io/en/develop/generated/colour.Lab_to_XYZ.html

By the way, I think you guys can take the `Colour` lib's document as an example
